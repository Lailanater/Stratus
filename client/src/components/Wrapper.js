import React from 'react';
import { useSelector } from 'react-redux';
import { ThemeProvider } from '@material-ui/styles';
import { SnackbarProvider } from 'notistack';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';

const Wrapper = props => {
  const theme = useSelector(state => state.theme);
  let currentTheme = createMuiTheme({
    palette: {
      type: theme,
      primary: { main: '#00838F' },
      secondary: { main: '#ed407a' }
    }
  });

  return (
    <ThemeProvider theme={currentTheme}>
      <SnackbarProvider maxSnack={2}>{props.children}</SnackbarProvider>
    </ThemeProvider>
  );
};

export default Wrapper;
