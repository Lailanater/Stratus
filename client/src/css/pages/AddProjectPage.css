#selectDirectoryButton {
  margin-top: 22px;
  padding: 10px;
  margin-left: 10px;
}

#project-path-input {
  width: 500px;
}

#project-name-input {
  width: 350px;
}
