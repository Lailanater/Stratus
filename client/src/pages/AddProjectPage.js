import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { Paper, Typography } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { addProject } from '../redux/actions/actionCreators';
import { Redirect } from 'react-router-dom';
import { withSnackbar } from 'notistack';
import { objectsAreEqual } from '../utils/helpers';
import styled from 'styled-components';
import isElectron from 'is-electron';

const PathTextInput = styled(TextField)`
  width: 500px;
`;

const NameTextInput = styled(TextField)`
  width: 350px;
`;

const AddProjectPage = props => {
  const [canRedirect, setCanRedirect] = useState(false);
  const projects = useSelector(state => state.projects);
  const dispatch = useDispatch();

  function setDefaultProjectName() {
    console.log(document.querySelector('#project-name-input').value);
    document.querySelector(
      '#project-name-input'
    ).value = getTextAfterLastSlash();
  }

  function replaceBacklashesWithForwardSlashes(path) {
    if (path.indexOf('/') === -1) {
      path = path.replace(/\\/g, '/');
    }
    return path;
  }

  function getTextAfterLastSlash() {
    let path = document.querySelector('#project-path-input').value;
    path = replaceBacklashesWithForwardSlashes(path);

    let slashIndex = path.lastIndexOf('/');

    return path.substr(slashIndex + 1, path.length);
  }

  function isValidPath(path) {
    if (!isElectron()) {
      return true; // Cannot validate the path. Assume given is valid.
    } else {
      const { remote } = window.require('electron');
      const fs = remote.require('fs');
      return fs.existsSync(path);
    }
  }

  function projectAlreadyExists(projectName, projectPath) {
    const projectToAdd = {
      name: projectName,
      path: projectPath
    };

    for (let i = 0; i < projects.length; i++) {
      let project = projects[i];
      if (objectsAreEqual(project, projectToAdd)) {
        return true;
      }
    }
    return false;
  }

  function handleOnClick() {
    const projectName = document.querySelector('#project-name-input').value;
    const projectPath = document.querySelector('#project-path-input').value;

    if (projectName === '' || projectPath === '') {
      props.enqueueSnackbar('Please update both fields first', {
        variant: 'error',
        autoHideDuration: 2000
      });
    } else {
      if (isValidPath(projectPath)) {
        if (!projectAlreadyExists(projectName, projectPath)) {
          dispatch(addProject(projectName, projectPath));
          setCanRedirect(true);
        } else {
          props.enqueueSnackbar('The project you entered already exists', {
            variant: 'error',
            autoHideDuration: 2000
          });
        }
      } else {
        props.enqueueSnackbar('The path you have entered is not valid', {
          variant: 'error',
          autoHideDuration: 2000
        });
      }
    }
  }

  function selectFolder() {
    if (isElectron()) {
      const options = {
        properties: ['openDirectory']
      };

      const { remote } = window.require('electron');
      remote.dialog.showOpenDialog(options).then(result => {
        let selectedDirectory = result.filePaths[0];
        if (selectedDirectory === undefined) {
          selectedDirectory = '';
        }

        document.querySelector(
          '#project-path-input'
        ).value = replaceBacklashesWithForwardSlashes(selectedDirectory);
        document.querySelector(
          '#project-name-input'
        ).value = getTextAfterLastSlash();
      });
    } else {
      alert('This function is not available on the web.');
    }
  }

  if (canRedirect) {
    return <Redirect to="/editProjects" />;
  }

  return (
    <div className="container">
      <Paper className="container">
        <Typography variant="h6">Project Path</Typography>
        <PathTextInput
          id="project-path-input"
          margin="normal"
          variant="outlined"
          onChange={setDefaultProjectName}
        />

        <Button
          style={{ marginTop: '22px', padding: '10px', marginLeft: '10px' }}
          variant="contained"
          onClick={selectFolder}
        >
          Select directory...
        </Button>

        <br />

        <Typography variant="h6">Project Name</Typography>
        <NameTextInput
          id="project-name-input"
          margin="normal"
          variant="outlined"
        />
      </Paper>

      <br />

      <Button variant="contained" color="secondary" onClick={handleOnClick}>
        Add Project
      </Button>
    </div>
  );
};

export default withSnackbar(AddProjectPage);
