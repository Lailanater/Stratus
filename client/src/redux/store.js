import { createStore } from 'redux';
import rootReducer from './reducers';
import { loadState } from '../localStorage';

let initialState = loadState();

if (initialState === undefined) {
  initialState = {
    isAppMenuDisplayed: false,
    theme: 'dark',
    currentProject: '',
    projects: [],
    menuName: '',
    needsRepeat: false,
    needsDtmf: false,
    needsVoice: false,
    dtmfOptions: [],
    defaultRouteTo: 'menu',
    isNextButtonDisabled: false,
    isFinishButtonDisabled: false
  };
}

export default createStore(rootReducer, initialState);
