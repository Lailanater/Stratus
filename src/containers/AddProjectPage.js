import React, {Component} from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import {Paper, Typography} from "@material-ui/core";

class AddProjectPage extends Component {
    constructor(props) {
        super(props);

        this.setDefaultProjectName = this.setDefaultProjectName.bind(this);
    }


    setDefaultProjectName() {
        console.log(document.querySelector("#project-name-input").value);
        document.querySelector("#project-name-input").value = this.getTextAfterLastBackSlash();
    }

    getTextAfterLastBackSlash() {
        let path = document.querySelector("#project-path-input").value;
        let backslashIndex = path.lastIndexOf("\\");

        return path.substr(backslashIndex + 1, path.length);
    }

    render() {

        return (
            <Paper>
                <Typography variant="h6">
                    Project Path
                </Typography>
                <TextField
                    id="project-path-input"
                    margin="normal"
                    variant="outlined"
                    onChange={this.setDefaultProjectName}
                />

                <Button variant="contained">
                    Select directory...
                </Button>

                <br />

                <Typography variant="h6">
                    Project Name
                </Typography>
                <TextField
                    id="project-name-input"
                    margin="normal"
                    variant="outlined"
                />

                <br />

                <Button variant="contained" color="secondary">
                    Add Project
                </Button>
            </Paper>
        );
    }
}

export default AddProjectPage;