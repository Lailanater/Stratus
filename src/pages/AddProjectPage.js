import React from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { Paper, Typography } from "@material-ui/core";
import { useDispatch } from "react-redux";
import { addProject } from "../redux/actions/actionCreators";
import { Link } from "react-router-dom";

const {remote} = window.require("electron");

const AddProjectPage = (props) => {

    const dispatch = useDispatch();

    function setDefaultProjectName() {
        console.log(document.querySelector("#project-name-input").value);
        document.querySelector("#project-name-input").value = getTextAfterLastBackSlash();
    }

    function getTextAfterLastBackSlash() {
        let path = document.querySelector("#project-path-input").value;
        let backslashIndex = path.lastIndexOf("\\");

        return path.substr(backslashIndex + 1, path.length);
    }

    function handleOnClick() {
        const projectName = document.querySelector("#project-name-input").value;
        const projectPath = document.querySelector("#project-path-input").value;
        dispatch(addProject(projectName, projectPath));
    }

    function selectFolder() {
        const options = {
            properties: ["openDirectory"]
        };

        remote.dialog.showOpenDialog(options).then(result => {
                const selectedDirectory = result.filePaths[0];

                document.querySelector("#project-path-input").value = selectedDirectory;
                document.querySelector("#project-name-input").value = getTextAfterLastBackSlash();
            }
        );
    }

    return (
        <Paper>
            <Typography variant="h6">
                Project Path
            </Typography>
            <TextField
                id="project-path-input"
                margin="normal"
                variant="outlined"
                onChange={setDefaultProjectName}
            />

            <Button variant="contained" onClick={selectFolder}>
                Select directory...
            </Button>

            <br />

            <Typography variant="h6">
                Project Name
            </Typography>
            <TextField
                id="project-name-input"
                margin="normal"
                variant="outlined"
            />

            <br />

            <Link to="/">
                <Button variant="contained" color="secondary" onClick={handleOnClick}>
                    Add Project
                </Button>
            </Link>
        </Paper>
    );
};

export default AddProjectPage;