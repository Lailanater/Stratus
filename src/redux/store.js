import {createStore} from "redux";
import rootReducer from "./reducers";

export const initialState = {
    isAppMenuDisplayed: false,
    theme: "dark",
    currentProject: "",
    projects: []
};

export default createStore(rootReducer);