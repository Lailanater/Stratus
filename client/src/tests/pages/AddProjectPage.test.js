import React from 'react';
import renderer from 'react-test-renderer';
import AddProjectPage from '../../pages/AddProjectPage';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Wrapper from '../../components/Wrapper';

it('renders without crashing', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Wrapper>
        <AddProjectPage />
      </Wrapper>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
