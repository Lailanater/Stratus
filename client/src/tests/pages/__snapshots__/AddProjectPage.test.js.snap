// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 container MuiPaper-rounded"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
    >
      Project Path
    </h6>
    <div
      className="MuiFormControl-root MuiTextField-root sc-bdVaJa kdjZLc MuiFormControl-marginNormal"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="project-path-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-100 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-101"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginLeft": "10px",
          "marginTop": "22px",
          "padding": "10px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Select directory...
      </span>
    </button>
    <br />
    <h6
      className="MuiTypography-root MuiTypography-h6"
    >
      Project Name
    </h6>
    <div
      className="MuiFormControl-root MuiTextField-root sc-bwzfXH iBveIV MuiFormControl-marginNormal"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="project-name-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-100 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-101"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <br />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Add Project
    </span>
  </button>
</div>
`;
